<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SCRuB Tutorial</title>

<script src="tutorial_files/header-attrs-2.14/header-attrs.js"></script>
<script src="tutorial_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="tutorial_files/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="tutorial_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="tutorial_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="tutorial_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="tutorial_files/navigation-1.1/tabsets.js"></script>
<link href="tutorial_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="tutorial_files/highlightjs-9.12.0/highlight.js"></script>
<link href="tutorial_files/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="tutorial_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SCRuB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fab fa-r-project"></span>
     
    Package
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="R-inputs.html">Inputs &amp; Outputs</a>
    </li>
    <li>
      <a href="tutorial.html">Tutorial</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="glyphicon glyphicon-zoom-in"></span>
     
    QIIME2 Plugin
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qiime2-intro.html">Inputs &amp; Outputs</a>
    </li>
    <li>
      <a href="Plasma-Tutorial-QIIME2-CLI.html">Command line interface</a>
    </li>
    <li>
      <a href="Plasma-Tutorial-QIIME2-API.html">Qiime2 interface</a>
    </li>
    <li>
      <a href="Plasma-Tutorial-QIIME2-standalone-API.html">Python interface</a>
    </li>
  </ul>
</li>
<li>
  <a href="prescrub_intro.html">
    <span class="fa fa-object-group"></span>
     
    preSCRuB
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Related documents
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="www.google.com">SCRuB Manuscript</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="plans.html">
    <span class="fa fa-sparkles"></span>
     
    Future developments
  </a>
</li>
<li>
  <a href="contribute.html">Contribute to SCRuB!</a>
</li>
<li>
  <a href="https://github.com/korem-lab/SCRuB/issues">
    <span class="fa fa-comment"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/korem-lab/SCRuB/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCRuB Tutorial</h1>

</div>


<style>
    body { background-color: #162138;
            text-color: whitesmoke}
</style>
<p><img src='images/SCRuB_logo.png' align="right" height="139" /></p>
<p>In this tutorial, we demonstrate how SCRuB should be used to remove
contamination from microbial samples. All you need to run SCRuB is a
count matrix of control and non-control samples. To run the SCRuB with
its full spatial component, which is strongly recommended, the well
locations of each sample must also be provided.</p>
<hr />
<div id="section" class="section level1 tabset">
<h1 class="tabset"></h1>
<div id="loading-the-data" class="section level2">
<h2>Loading the data</h2>
<p>The starting point for SCRuB is a matrix, in which each row
represents a sample, and each column represents each sample’s read
counts that correspond to a certain taxa (e.g ASV, OTU, species). In
general, we recommend to run SCRuB on the most granular version of the
data possible, and so data should only be grouped to higher orders of
phylogeny (e.g. genus, family…) <em>after</em> completing the SCRuB
pipeline.</p>
<p>We start by loading the necessary packages, such as
<code>SCRUB</code>. We also include the use of <code>tidyverse</code>
within our tutorial</p>
<pre class="r"><code>set.seed(1)
library(tidyverse)
library(SCRuB)</code></pre>
<p>As an example dataset, we use processed samples publicly available
from Qiita (<strong>link</strong>), from a dataset of plasma samples,
made public by Poore et al. In the ensuing pipeline, we demonstrate how
SCRuB can be used to remove the contamination from the plasma using the
control samples, and pub liscly available metadata published by Poore et
al.</p>
<pre class="r"><code>data &lt;- read.csv(&#39;plasma_data.csv&#39;, row.names=1) %&gt;% as.matrix()
dim(data)</code></pre>
<pre><code>## [1]  381 7777</code></pre>
<p>Next, we load a metadta file, which is in the format required for
<code>SCRuB</code>. Before running SCRuB, we recommend that users make
sure their metadata file formats are aligned with the one shown
here.</p>
<pre class="r"><code>metadata &lt;- read.csv(&#39;plasma_metadata.csv&#39;, row.names=1)
metadata %&gt;% head()</code></pre>
<pre><code>##              is_control sample_type sample_well
## 12692.150003      FALSE      plasma         H14
## 12692.150005      FALSE      plasma         L14
## 12692.150006      FALSE      plasma         B16
## 12692.150007      FALSE      plasma         J14
## 12692.150011      FALSE      plasma         J16
## 12692.150013      FALSE      plasma         L16</code></pre>
</div>
<div id="running-scrub" class="section level2">
<h2>Running SCRuB</h2>
<pre class="r"><code>scr_out &lt;- SCRuB( data[,1:750], metadata ) # shortening feature space to shorten tutorial runtime...</code></pre>
<pre><code>## [1] &quot;Incorporating the well metadata to track well-to-well leakage!&quot;
## [1] &quot;SCRuBbing away contamination in the control blank DNA extraction controls...&quot;
## [1] &quot;SCRuBbing away contamination in the control blank library prep controls...&quot;</code></pre>
</div>
<div id="evaluating-outputs" class="section level2">
<h2>Evaluating outputs</h2>
<p>Let’s take a look at the results. The estimated level of
contamination was resonable, as the fitted <code>p</code> parameters
indicate our samples are typically close to <code>3%</code>
contamination.</p>
<pre class="r"><code>scr_out$p %&gt;% boxplot()</code></pre>
<p><img src="tutorial_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>To take a look at the cleaned samples, refer to the
<code>decontaminated_samples</code> entry.</p>
<pre class="r"><code>decontaminated_samples &lt;- scr_out$decontaminated_samples
decontaminated_samples[1:10, 25:40]</code></pre>
<pre><code>##              G000006425 G000006445 G000006515 G000006565 G000006605 G000006685
## 12692.150003          0          0          0        305          0          0
## 12692.150005          0          0          0        441          0          0
## 12692.150006          0          0          0      11235          0          0
## 12692.150007          0          0          0       3888          0          0
## 12692.150011          0          0          0         33          2          0
## 12692.150013          0          0          0        215          0          0
## 12692.150022          0          7          0        797          0          0
## 12692.150031          0          0          0       9027          0          0
## 12692.150032          0          0          0       1436          0          0
## 12692.150080          0          0          0        223          0          0
##              G000006725 G000006765 G000006925 G000006945 G000007065 G000007145
## 12692.150003          0         26          0          0          0          0
## 12692.150005          0         22          1          0          0          0
## 12692.150006          0        142          0          0          0         30
## 12692.150007          0         57          0          0          0          0
## 12692.150011          0         23          0          0          0          4
## 12692.150013          0         38          0          0          0          0
## 12692.150022          0         26          0          0          0          0
## 12692.150031          0         60          0          0          0          2
## 12692.150032          0         55          0          0          0        274
## 12692.150080          0         50          0          0          0          4
##              G000007205 G000007225 G000007345 G000007605
## 12692.150003          0          0          0          0
## 12692.150005          0          0          0          0
## 12692.150006          0          0          0          0
## 12692.150007          0          0          0          0
## 12692.150011          0          0          0          0
## 12692.150013          0          0          0          0
## 12692.150022          0          0          0          0
## 12692.150031          0          0          0          0
## 12692.150032          0          0          0          0
## 12692.150080          0          0          0          0</code></pre>
<p>An output from ever inner iteration of the SCRUB function is the
<code>gamma</code> parameter, which is a vector representing the
estimated relative abundance of a contamination community.</p>
<pre class="r"><code>scr_out$inner_iterations$`control blank library prep`$gamma %&gt;% plot()</code></pre>
<p><img src="tutorial_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>It is strongly recommended to incorporate the well metadata of each
sample into SCRuB, as this makes it possible ot directly account for
potential well leakage into negative controls, reducing the risk of
wrongly removing certain species during decontamination.</p>
<p>One additional output when incoprorating the well metadata is the
<code>alpha</code> parameters, which is a <code>n_control</code> by
<code>n_samples + 1</code> matrix that represents the estimated level of
leakage into each control. The last column of the <code>alpha</code>
matrix represents the estimated proportion of each control that
originates from the contamination source. In this case,
<code>SCRuB</code> estimated a very high level of leakage for many DNA
extraction samples, with a median estimated well-to-well leakage of
<code>60%</code>.</p>
<pre class="r"><code>boxplot( scr_out$inner_iterations$`control blank DNA extraction`$alpha[, ncol(scr_out$inner_iterations$`control blank DNA extraction`$alpha)] )</code></pre>
<p><img src="tutorial_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
